addons/pos_pinelabs/static/src/components/payment_screen/payment_screen.js

```diff
@@ -0,0 +1,28 @@
+import { PaymentScreen } from "@point_of_sale/app/screens/payment_screen/payment_screen";
+import { patch } from "@web/core/utils/patch";
+import { onMounted } from "@odoo/owl";
+
+patch(PaymentScreen.prototype, {
+    setup() {
+        super.setup(...arguments);
+        onMounted(async () => {
+            const pendingPaymentLine = this.currentOrder.payment_ids.find(
+                (paymentLine) =>
+                    paymentLine.payment_method_id.use_payment_terminal === "pinelabs" &&
+                    !paymentLine.is_done() &&
+                    paymentLine.get_payment_status() !== "pending"
```

Comment:
the `pendingPaymentLine` is a line that has as status anything but "pending" ??

Reply:
Hello @vlst-odoo 

In the `pendingPaymentLine`, a payment line signifies that a payment request has been initiated, but the payment has not yet been completed. When a payment line is created from the payment method, it enters a `pending` status. After the payment request has been initiated, it transitions to the `waitingCard` state.

We are managing a case where, upon returning to the `PaymentScreen` from the `ProductScreen` or any other screen, we update the status of the pending payment line.

Reply:
the point is simple: having a variable called `pendingPaymentLine` for a line that has status something other than `pending` is confusing. You can just rename the variable.

Reply:
Do you have any suggestions for a name since the line is not in a pending state and is yet to be completed? I am also confused about what to call it. The current state of the payment line is labeled as `waitingCard`. Could we consider renaming it to `waitingPaymentLine` or something similar for clarity?

Reply:
i think `waitingPaymentLine` works for now